Tests fail with mawk

Some awk code was written expecting gawk to be installed, but in reality we only have mawk:

```
FAIL: test_enable_disable_needs_root (test_script.UbuntuAdvantageScriptTest)
The script must be run as root for enable and disable actions.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<BUILDDIR>>/ubuntu-advantage-tools-13+82+git.291ae41~ubuntu18.04.1/tests/test_script.py", line 18, in test_enable_disable_needs_root
    self.assertIn('This command must be run as root', process.stderr)
AssertionError: 'This command must be run as root' not found in 'awk: line 3: syntax error at or near ,\nawk: line 7: syntax error at or near ,\n'
```
I can reproduce this in an lxd if I remove gawk and leave just mawk installed.

--- a/modules/service.sh
+++ b/modules/service.sh
@@ -12,7 +12,7 @@ service_from_command() {
 
      echo "$command" | sed -n -r \
          "s,^is-([^-]+)-enabled$,\1,p;\
-          s,^(enable|disabled)-([^-]+)$,\2,p"
+          s,^(enable|disable)-([^-]+)$,\2,p"
 }
 
 service_enable() {

--- a/modules/service.sh
+++ b/modules/service.sh
@@ -11,8 +11,8 @@ service_from_command() {
     local command="$1"
 
      echo "$command" | sed -n -r \
-         "s,^is-([^-]+)-enabled$,\1,p;\
-          s,^(enable|disable)-([^-]+)$,\2,p"
+         "s,^is-(.+)-enabled$,\1,p;\
+          s,^(enable|disable)-(.+)$,\2,p"
 }
 
 service_enable() {

--- a/modules/service.sh
+++ b/modules/service.sh
@@ -10,16 +10,9 @@ check_user() {
 service_from_command() {
     local command="$1"
 
-     echo "$command" | awk '
-         /^is-.*-enabled$/ {
-             match($0, /^is-(.*)-enabled$/, m);
-             print m[1];
-         }
-         /^(enable|disable)-/ {
-             match($0, /^(enable|disable)-(.*)/, m);
-             print m[2];
-         }
-    '
+     echo "$command" | sed -n -r \
+         "s,^is-([^-]+)-enabled$,\1,p;\
+          s,^(enable|disabled)-([^-]+)$,\2,p"
 }
 
 service_enable() {

--- a/modules/service.sh
+++ b/modules/service.sh
@@ -11,8 +11,8 @@ service_from_command() {
     local command="$1"
 
      echo "$command" | sed -n -r \
-         's,^is-(.+)-enabled$,\1,p;'\
-          's,^(enable|disable)-(.+)$,\2,p'
+         's,^is-(.+)-enabled$,\1,p;
+          s,^(enable|disable)-(.+)$,\2,p'
 }
 
 service_enable() {

--- a/modules/service.sh
+++ b/modules/service.sh
@@ -11,8 +11,8 @@ service_from_command() {
     local command="$1"
 
      echo "$command" | sed -n -r \
-         "s,^is-(.+)-enabled$,\1,p;\
-          s,^(enable|disable)-(.+)$,\2,p"
+         's,^is-(.+)-enabled$,\1,p;'\
+          's,^(enable|disable)-(.+)$,\2,p'
 }
 
 service_enable() {

